framework/physics/SolverBase/chi_solver.cc

```diff
@@ -64,6 +72,10 @@ Solver::Solver(const chi::InputParameters& params)
     time_step_controller_ = Chi::GetStackItemPtrAsType<TimeStepController>(
       Chi::object_stack, handle, __FUNCTION__);
   }
+
+  ChiInvalidArgumentIf(has_controller and has_time_controls,
+                       "Both \"has_controller\" and \"has_time_controls\" "
+                       "cannot be specified at the same time.");
```

Comment:
Unnecessary if you just require a controller handle.

Reply:
But thats the whole point. We don't just require a handle. We can accept just the basic controls.